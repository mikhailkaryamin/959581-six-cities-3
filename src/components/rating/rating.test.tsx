import React from "react";
import renderer from "react-test-renderer";
import {
  ClassModificator
} from "../../consts";
import Rating from './rating';

describe(`Rating render correctly`, () => {
  test(`Rating place card render correctly`, () => {
    const tree = renderer
      .create(
          <Rating
            rating={3}
            classModificator={ClassModificator.PLACE_CARD}
          />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Rating property render correctly`, () => {
    const tree = renderer
      .create(
          <Rating
            rating={3}
            classModificator={ClassModificator.PROPERTY}
          />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

