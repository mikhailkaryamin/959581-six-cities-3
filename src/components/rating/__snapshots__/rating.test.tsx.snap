// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating render correctly Rating place card render correctly 1`] = `
<div
  className="place-card__rating rating"
>
  <div
    className="place-card__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating render correctly Rating property render correctly 1`] = `
<div
  className="property__rating rating"
>
  <div
    className="property__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
  <span
    className="property__rating-value rating__value"
  >
    3
  </span>
</div>
`;
